src/Microsoft.AspNet.Mvc.Core/Filters/ProducesAttribute.cs

```diff
@@ -37,11 +39,29 @@ public override void OnResultExecuting([NotNull] ResultExecutingContext context)
 
         private List<MediaTypeHeaderValue> GetContentTypes(string firstArg, string[] args)
```

Comment:
This code is repeated twice, we have avoided adding helpers till now, any suggestions?